client/components/email-verification/email-verification-dialog/index.jsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import noop from 'lodash/noop';
+import i18n from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import Spinner from 'components/spinner';
+import { getCurrentUserEmail } from 'state/current-user/selectors';
+import {
+	verifyEmail,
+	resetVerifyEmailState,
+} from 'state/current-user/email-verification/actions';
+
+class VerifyEmailDialog extends Component {
+
+	getResendButtonLabel() {
+		if ( this.props.emailSent || this.props.error ) {
+			return i18n.translate( 'Email Sent' );
```

Comment:
Kind of. The component never indicated an error state before, even if the email wasn't sent correctly. Didn't want to make a change to the behaviour in addition to the rest of the changes. It may warrant a PR in the future or at least an issue filed, what do you think?